### PR TITLE
feat(live): add a new resource to manage notification cofiguration

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a new resource to manage notification cofiguration.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support a new resource to manage notification cofiguration.
2. support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/live" TESTARGS="-run TestAccNotificationConfiguration_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/live -v -run TestAccNotificationConfiguration_basic -timeout 360m -parallel 4
=== RUN   TestAccNotificationConfiguration_basic
=== PAUSE TestAccNotificationConfiguration_basic
=== CONT  TestAccNotificationConfiguration_basic
--- PASS: TestAccNotificationConfiguration_basic (305.13s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/live      305.174s
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<
image
    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<
image

### CODE REVIEW

File: huaweicloud/services/live/resource_huaweicloud_live_notification_cofiguration.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+	_, err := client.Request("PUT", notificationPath, &notificationOpt)
+	return err
+}
+
+func buildNotificationCofigBodyParams(d *schema.ResourceData) map[string]interface{} {
+	params := map[string]interface{}{
+		"url":            d.Get("url"),
+		"auth_sign_key":  d.Get("auth_sign_key"),
+		"call_back_area": d.Get("call_back_area"),
```
Comment (jinyangyang222) [inline huaweicloud/services/live/resource_huaweicloud_live_notification_cofiguration.go:111]:
```suggestion
		"auth_sign_key":  utils.ValueIgnoreEmpty(d.Get("auth_sign_key")),
		"call_back_area": utils.ValueIgnoreEmpty(d.Get("call_back_area")),
```

Comment (ruwenqiang123) [reply huaweicloud/services/live/resource_huaweicloud_live_notification_cofiguration.go:113]: This parameter can be left blank.

Comment (ruwenqiang123) [reply huaweicloud/services/live/resource_huaweicloud_live_notification_cofiguration.go:113]: This parameter can be left blank.

File: docs/resources/live_notification_cofiguration.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  Changing this parameter will create a new resource.
+
+* `domain_name` - (Required, String, ForceNew) Specifies the ingest domain name to which the notification cofiguration
+  belongs.
+  Changing this parameter will create a new resource.
+
+* `url` - (Required, String) Specifies the callback URL, which must start with `http://` or `https://`.
+
+* `auth_sign_key` - (Optional, String) Specifies the authentication key.
+  The valid length is `32` to `128` characters. This parameter can be left blank.
```
Comment (deer-hang) [inline docs/resources/live_notification_cofiguration.md]:
```suggestion
  The valid length is `32` to `128` characters.
```

File: docs/resources/live_notification_cofiguration.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+* `call_back_area` - (Optional, String) Specifies the area where the server for receiving callback notification
```
Comment (deer-hang) [inline docs/resources/live_notification_cofiguration.md]:
```suggestion
* `call_back_area` - (Optional, String) Specifies the region where the server that receives callback notifications is located.
```

File: docs/resources/live_notification_cofiguration.md (hunk 2 above)
Comment (deer-hang) [reply docs/resources/live_notification_cofiguration.md]: `Optional` already indicates that this field can be left unconfigured, so no special instructions are needed.

File: docs/resources/live_notification_cofiguration.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  is located. This parameter can be left blank.
+  The valid vaules are as follows:
+  + **mainland_china**: Indicates Chinese mainland.
+  + **outside_mainland_china**: Indicates outside the Chinese mainlan.
```
Comment (deer-hang) [inline docs/resources/live_notification_cofiguration.md]:
```suggestion
  + **outside_mainland_china**: Indicates outside the Chinese mainland.
```

File: huaweicloud/services/live/resource_huaweicloud_live_notification_cofiguration.go [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+func createOrUpdateNotificationCofig(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
+	notificationHttpUrl := "v1/{project_id}/notifications/publish"
+	notificationPath := client.Endpoint + notificationHttpUrl
+	notificationPath = strings.ReplaceAll(notificationPath, "{project_id}", client.ProjectID)
+	notificationPath = fmt.Sprintf("%s?domain=%v", notificationPath, d.Get("domain_name"))
+
+	notificationOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildNotificationCofigBodyParams(d)),
```
Comment (deer-hang) [inline huaweicloud/services/live/resource_huaweicloud_live_notification_cofiguration.go]: It seems that `utils.RemoveNil` here has no actual meaning.

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/live" TESTARGS="-run TestAccNotificationConfiguration_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/live -v -run TestAccNotificationConfiguration_basic -timeout 360m -parallel 4
=== RUN   TestAccNotificationConfiguration_basic
=== PAUSE TestAccNotificationConfiguration_basic
=== CONT  TestAccNotificationConfiguration_basic
--- PASS: TestAccNotificationConfiguration_basic (257.30s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/live      257.354s
```

**Pull request merged**